### DISCUSSION TITLE
remark-highlight.js not rendering all classes

### DISCUSSION THREAD

**Discussion by Haniver** [remarkjs/remark#984] [Q&A] [answered]
I'm using remark-highlight.js with react-remark. I'm testing with the example in the [documentation](https://github.com/remarkjs/remark-highlight.js/tree/8d4895e):
```
~~~css
h1 {
  color: red;
}
~~~
```

That is supposed to render this HTML code:
```html
<pre><code class="hljs language-css"><span class="hljs-selector-tag">h1</span> {
  <span class="hljs-attribute">color</span>: red;
}</code></pre>
```

But I'm actually getting this:
``` html
<pre><code class="language-css">h1 {
  color: red;
}
</code></pre>
```

That is, the text is being formatted with the "language-css" class, but no coloring is present. Maybe the plugin is not active at all, but remark plugins in general are working because I'm also inserting some emojis and they render well with 'remark-gemoji.

My React code looks something like this:
``` javascript
import { useState, useEffect } from 'react'
import { Card, CardBody } from 'reactstrap'
import { useParams } from "react-router-dom"
import { useRemarkSync } from 'react-remark'
import remarkGemoji from 'remark-gemoji'
import authHeader from '../services/auth.header'
import axios from 'axios'
import customUrls from '../services/customUrls'

const Documentos = () => {

    const { id } = useParams()

    const [markdown, setMarkdown] = useState("Espera a que cargue el documento...")
    useEffect(async () => {
        const res = await axios({
            method: 'get',
            url: `${customUrls.ApiUrl()}markdown/${id}`,
            headers: authHeader()
        })
        setMarkdown(res.data)
    }, [id])
    const contenido = useRemarkSync(markdown, {remarkPlugins: [remarkHighlightjs, remarkParse,  remarkGemoji]})

    return (
        <Card>
            <CardBody>
                {contenido}
            </CardBody>
        </Card>
    )
}
export default Documentos
```

**Comment by wooorm** [comment]
Your code does not include a highlighting project.

**Comment by Haniver** [comment, marked as answer]
Yes, I made a mistake when pasting the code here on this post. The import is not missing in my original code.
What I did was to:
- Switch to react-markdown
- Then the hljs classes appeared in the code, but nothing was formatted because the css files to which they point were not imported (I have no idea why)
- So I grabbed a CSS I liked directly from node_modules/highlight.js/styles and put it in my source folders, required it in my component code and tada!
